### PR TITLE
feat(expressions): expressions language support and editor [KM-19]

### PR DESCRIPTION
**Pull request by sumimakito**
# Summary

This pull request aims to add shared components to support [Kong's Expressions language](https://docs.konghq.com/gateway/latest/reference/expressions-language/).

This package mainly provides `ExpressionsEditor.vue`–A `monaco-editor` based component that supports syntax highlighting and autocomplete for the Expressions language.

KM-19

### CODE REVIEW

File: packages/core/expressions/src/monaco.ts [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+
+      // split everything the user has typed on the current line up at each space, and only look at the last word
+      const lastChars = model.getValueInRange({
+        startLineNumber: position.lineNumber,
+        startColumn: 0,
+        endLineNumber: position.lineNumber,
+        endColumn: position.column,
+      })
+
+      // nosemgrep: javascript.lang.security.audit.incomplete-sanitization.incomplete-sanitization
```
Comment (Leopoldthecoder) [inline packages/core/expressions/src/monaco.ts:135]: nit: we can remove this
```suggestion
```

**Review by Leopoldthecoder — APPROVED** [review]
LGTM

File: packages/core/expressions/src/monaco.ts (hunk 1 above)
Comment (sumimakito) [reply packages/core/expressions/src/monaco.ts]: Thanks for reminding me of this. I've removed this line

**Review by Leopoldthecoder — APPROVED** [review]
LGTM

File: packages/core/expressions/src/monaco.ts [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+  const languageId = getLanguageId(schema.name)
+
+  if (monaco.languages.getEncodedLanguageId(languageId) !== 0) {
+    console.warn(`Language "${languageId}" has already been registered`)
+    return { languageId }
+  }
+
+  const tokenTree = buildTokenTree(schema.definition)
+
+  console.debug(`Registering language "${languageId}" with token tree:`, tokenTree)
```
Comment (kaiarrowood) [inline packages/core/expressions/src/monaco.ts:67]: question: do we want this?

File: packages/core/expressions/src/components/ExpressionsEditor.vue [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+    isParsingActive.value = true
+  }
+
+  if (editor !== undefined && editor.getValue() !== newExpression) {
+    editor.setValue(newExpression)
+  }
+})
+
+watch([expression, schema], (() => {
+  const cb = () => {
```
Comment (kaiarrowood) [inline packages/core/expressions/src/components/ExpressionsEditor.vue:117]: nit: could a more descriptive variable name be used or maybe add a comment explaining what this watch is doing?

File: packages/core/expressions/src/components/ExpressionsEditor.vue [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+    }
+  }
+
+  monaco.editor.setModelMarkers(editorModel, 'kong-expressions-editor', markers)
+})
+</script>
+
+<style lang="scss" scoped>
+.expression-editor {
+  border: 1px solid $kui-color-border;
```
Comment (kaiarrowood) [inline packages/core/expressions/src/components/ExpressionsEditor.vue:200]:
```suggestion
  border: $kui-border-width-10 solid $kui-color-border;
```

nit

File: packages/core/expressions/src/components/ExpressionsEditor.vue [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+  border-radius: 3px;
+  min-height: 200px;
+  overflow: hidden;
+  transition: border-color linear 150ms;
+  width: 100%;
+
+  &.invalid {
+    border-color: #d44324;
```
Comment (kaiarrowood) [inline packages/core/expressions/src/components/ExpressionsEditor.vue:208]: nit: is there a design token that can be used?

**Review by kaiarrowood — DISMISSED** [review]
Functionally this looks good to me. Nothing but nits on the code 🚀

**Review by Leopoldthecoder — APPROVED** [review]

**Pull request merged**